datashader/tests/test_transfer_functions.py

```diff
@@ -1049,57 +1049,35 @@ def test_shade_should_handle_zeros_array():
     assert img is not None
 
 
-def test_shade_with_discrete_color_key_numpy():
+def test_shade_with_discrete_color_key():
     data = np.array([[0, 0, 0, 0, 0],
                      [0, 1, 1, 1, 0],
                      [0, 2, 2, 2, 0],
                      [0, 3, 3, 3, 0],
                      [0, 0, 0, 0, 0]], dtype='uint32')
-    arr = tf.Image(data, dims=['x', 'y'])
-    img = tf.shade(arr, color_key={1: 'aqua', 2:'purple', 3:'yellow'})
-    assert img is not None
-
-
-def test_shade_with_discrete_color_key_dask():
-    data = np.array([[0, 0, 0, 0, 0],
-                     [0, 1, 1, 1, 0],
-                     [0, 2, 2, 2, 0],
-                     [0, 3, 3, 3, 0],
-                     [0, 0, 0, 0, 0]], dtype='uint32')
-    data = da.from_array(data)
-    arr = tf.Image(data, dims=['x', 'y'])
-    img = tf.shade(arr, color_key={1: 'aqua', 2:'purple', 3:'yellow'})
-    assert img is not None
-
-
-def test_shade_with_discrete_color_key_with_missing_values():
-    data = np.array([[0, 0, 0, 0, 0],
-                     [0, 1, 1, 1, 0],
-                     [0, 2, 2, 2, 0],
-                     [0, 3, 3, 3, 0],
-                     [0, 0, 0, 0, 0]], dtype='uint32')
-    arr = tf.Image(data, dims=['x', 'y'])
-    img = tf.shade(arr, color_key={1: 'aqua', 2:'purple', 3:'yellow', 4:'black'})
-    assert img is not None
-
-
-def test_shade_with_discrete_color_key_with_wrong_types():
-    data = np.array([[0, 0, 0, 0, 0],
-                     [0, 1, 1, 1, 0],
-                     [0, 2, 2, 2, 0],
-                     [0, 3, 3, 3, 0],
-                     [0, 0, 0, 0, 0]], dtype='uint32')
-    arr = tf.Image(data, dims=['x', 'y'])
-    img = tf.shade(arr, color_key={1: 'aqua', 2:'purple', 3:'yellow', 4.4:'black'})
-    assert img is not None
-
-
-def test_shade_with_discrete_color_key_with_float_arr():
-    data = np.array([[0, 0, 0, 0, 0],
-                     [0, 1, 1, 1, 0],
-                     [0, 2, 2, 2, 0],
-                     [0, 3, 3, 3, 0],
-                     [0, 0, 0, 0, 0]], dtype='f8')
-    arr = tf.Image(data, dims=['x', 'y'])
-    img = tf.shade(arr, color_key={1: 'aqua', 2:'purple', 3:'yellow'})
-    assert img is not None
+    color_key = {1: 'white', 2: 'purple', 3: 'yellow', 4.4: 'pink'}
```

Comment:
Thanks, I just updated some description.
4.4 here means that this also works with color keys with wrong types. Should it be eliminated?

Reply:
I don't think we should be testing for undefined behavior, and what happens for 4.4 in a uint seems like it should be treated as undefined.  I.e., what _should_happen in such a case?  Seems like that's a user error rather than a supported configuration.